app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -234,6 +234,8 @@ def parse_updates(vim, parser, updated_objects)
 
       retrieve_extra_props(managed_object, props)
 
+      clear_managed_object_connection!(managed_object)
+
       parser.parse(managed_object, update_kind, props)
```

Comment:
Another options is to pass `managed_object.class, managed_object._ref` in here instead of the whole object, need to go through the calls in the parser to ensure I'm not missing any other usages though